### ISSUE TITLE
Flatpak: directories mounted via `--filesystem=xdg-data/<app_name>:create` not handled by OpenURI

### ISSUE THREAD

**Issue by liushuyu** [flatpak/xdg-desktop-portal#594]
I am the current maintainer of https://github.com/flathub/org.yuzu_emu.yuzu. Our app uses `--filesystem=xdg-data/<app_name>:create` so that users using AppImage or distro distributed versions can easily switch to Flatpak without any hassle.

Now our app has a function to open the data folder in other apps (like Dolphin, Nautilus, etc, see below for a screenshot) so that users can modify the files within. However, the function does not work and `xdg-desktop-portal` will report `XDP: Rejecting open request as fd has no path associated to it`.

I think my issue is similar to but also a bit different from #591 #592 which deals with mounts but in different locations.

There is also a minimal reproducer if you don't want to install our app to reproduce this issue:

1. Find a random Flatpak app and execute `flatpak override <app_id> --filesystem=xdg-data/mount-test:create`
2. Execute `flatpak run --command=sh <app_id>`
3. Execute in the Flatpak shell: `xdg-open ~/.var/app/<app_id>/data/mount-test/`

If you want to use our app to reproduce the issue:

![screenshot](https://user-images.githubusercontent.com/6829345/122131285-2fa62a80-cdf6-11eb-9eac-38caa043bcc9.png)

**Comment by smcv** [comment]
> I think my issue is similar to but also a bit different from #591 #592 which deals with mounts but in different locations.

Yes, I think you're right. Like #591 and the /var/cache, /var/tmp variations of #592, this is going to need some refactoring, because we can't tell what path on the host we need to be looking at from just the `realpath()` of the path in the container: we'll need to try several possibilities.
